File: app/src/main/java/de/eknoes/inofficialgolem/SettingsActivity.java
package de.eknoes.inofficialgolem;


import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceActivity;
import android.preference.PreferenceManager;
import android.view.View;

import static de.eknoes.inofficialgolem.ArticleFragment.*;

public class SettingsActivity extends PreferenceActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.preferences);

        findPreference("how_to_key").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                Intent intent = new Intent(SettingsActivity.this, ArticleView.class);
                intent.putExtra(ARTICLE_URL, "https://account.golem.de/product/subscription");
                intent.putExtra(FORCE_WEBVIEW, true);
                intent.putExtra(NO_ARTICLE, true);
                startActivity(intent);
                return true;
            }
        });

        findPreference("choose_layout").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                Intent intent = new Intent(SettingsActivity.this, ArticleView.class);
                intent.putExtra(ARTICLE_URL, "https://www.golem.de/sonstiges/ansicht/");
                intent.putExtra(FORCE_WEBVIEW, true);
                intent.putExtra(NO_ARTICLE, true);
                startActivity(intent);
                return true;
            }
        });

        findPreference("privacy").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                Intent intent = new Intent(SettingsActivity.this, ArticleView.class);
                intent.putExtra(ARTICLE_URL, "https://projekte.eknoes.de/datenschutz.html");
                intent.putExtra(FORCE_WEBVIEW, true);
                intent.putExtra(NO_ARTICLE, true);
                startActivity(intent);
                return true;
            }
        });

        findPreference("join_beta").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                Intent webIntent = new Intent(Intent.ACTION_VIEW);
                webIntent.setData(Uri.parse("https://play.google.com/apps/testing/de.eknoes.inofficialgolem"));
                startActivity(webIntent);
                return true;
            }
        });

        findPreference("start_contact").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                /*Intent emailIntent = new Intent(Intent.ACTION_SENDTO, Uri.fromParts(
                        "mailto","projekte@eknoes.de", null));
                emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Inofficial golem.de Reader");
                startActivity(Intent.createChooser(emailIntent, getResources().getString(R.string.sendMail)));*/
                Intent webIntent = new Intent(Intent.ACTION_VIEW);
                webIntent.setData(Uri.parse("https://github.com/eknoes/golem-android-reader/issues"));
                startActivity(webIntent);
                return true;
            }
        });
    }
}

File: app/src/main/java/de/eknoes/inofficialgolem/SettingsActivity.kt
package de.eknoes.inofficialgolem

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.EditTextPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceManager
import de.eknoes.inofficialgolem.ArticleFragment.FORCE_WEBVIEW
import de.eknoes.inofficialgolem.ArticleFragment.NO_ARTICLE


class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.settings, SettingsFragment())
                .commit()
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.preferences, rootKey)

            val aboKey = findPreference<EditTextPreference>("abo_key")

            aboKey?.setOnPreferenceChangeListener { _: Preference, _: Any ->
                PreferenceManager.getDefaultSharedPreferences(context).edit() .putLong("last_refresh", 0).apply()
                true
            }
        }

        override fun onPreferenceTreeClick(preference: Preference?): Boolean {
            if (preference != null) {
                if (preference.key == "choose_layout") {
                    val intent = Intent(context, ArticleView::class.java)
                    intent.setData(Uri.parse("https://www.golem.de/sonstiges/ansicht/"))
                    intent.putExtra(FORCE_WEBVIEW, true)
                    intent.putExtra(NO_ARTICLE, true)
                    startActivity(intent)
                    return true
                } else if (preference.key == "how_to_key") {
                    val intent = Intent(context, ArticleView::class.java)
                    intent.setData(Uri.parse( "https://account.golem.de/product/subscription"))
                    intent.putExtra(FORCE_WEBVIEW, true)
                    intent.putExtra(NO_ARTICLE, true)
                    startActivity(intent)
                    return true
                } else if (preference.key == "how_to_darkmode") {
                    val intent = Intent(context, ArticleView::class.java)
                    intent.setData(Uri.parse( "https://account.golem.de/product/subscription#videotype1"))
                    intent.putExtra(FORCE_WEBVIEW, true)
                    intent.putExtra(NO_ARTICLE, true)
                    startActivity(intent)
                    return true
                } else if (preference.key == "github") {
                    val intent = Intent(Intent.ACTION_VIEW)
                    intent.data = Uri.parse("https://github.com/eknoes/golem-android-reader/issues")
                    startActivity(intent)
                    return true
                } else if (preference.key == "start_contact") {
                    val intent = Intent(Intent.ACTION_SENDTO).apply {
                        data = Uri.parse("mailto:projekte@eknoes.de")
                        putExtra(Intent.EXTRA_SUBJECT, "Golem.de Android App")
                        putExtra(Intent.EXTRA_TEXT, "Hi Sönke,")
                    }
                    startActivity(intent)
                    return true
                } else if (preference.key == "join_beta") {
                    val intent = Intent(Intent.ACTION_VIEW)
                    intent.data = Uri.parse("https://play.google.com/apps/testing/de.eknoes.inofficialgolem")
                    startActivity(intent)
                    return true
                } else if (preference.key == "privacy") {
                    val intent = Intent(context, ArticleView::class.java)
                    intent.setData(Uri.parse("https://projekte.eknoes.de/datenschutz.html"))
                    intent.putExtra(FORCE_WEBVIEW, true)
                    intent.putExtra(NO_ARTICLE, true)
                    startActivity(intent)
                    return true
                } else if (preference.key == "open_imprint") {
                    val intent = Intent(context, ArticleView::class.java)
                    intent.setData(Uri.parse("https://www.golem.de/sonstiges/impressum.html"))
                    startActivity(intent)
                    return true
                }

            }

            return super.onPreferenceTreeClick(preference)
        }
        
    }
}